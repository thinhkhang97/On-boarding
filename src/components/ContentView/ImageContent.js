// @flow
// Still have bug with android devices
import * as React from 'react';
import { View, StyleSheet, Image, Dimensions } from 'react-native';
import AvatarView from './AvatarView';

type Props = {
  source: {
    uri: string,
  },
  style?: View.propTypes.style,
  avatar?: boolean,
  imageStyle?: View.propTypes.style,
  avatarBorderColor?: string,
};

const screenWidth = Dimensions.get('window').width;

class ImageContent extends React.Component<Props> {
  static defaultProps = {
    avatar: false,
    style: {},
    imageStyle: {},
  };

  render() {
    const { source, avatar, style, imageStyle, avatarBorderColor } = this.props;
    return (
      <View
        style={[
          !avatar && {
            width: '100%',
            height: '100%',
            paddingTop: 30,
          },
          style,
        ]}
      >
        {avatar ? (
          <AvatarView borderColor={avatarBorderColor} source={source} />
        ) : (
          <Image
            style={[
              {
                width: 'auto',
                height: '100%',
              },
              imageStyle,
            ]}
            source={source}
            resizeMode="contain"
          />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({});

export default ImageContent;
