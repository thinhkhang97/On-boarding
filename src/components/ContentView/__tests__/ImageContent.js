import * as React from 'react';
import renderer from 'react-test-renderer';
import ImageContent from '../ImageContent';

test('Render avatar view correctly', ()=>{
    const testImageContent = renderer.create(<ImageContent/>).toJSON();
    expect(testImageContent).toMatchSnapshot();
})