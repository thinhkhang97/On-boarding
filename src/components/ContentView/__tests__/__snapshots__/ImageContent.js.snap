// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avatar view correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingTop": 30,
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {
          "height": "100%",
          "width": "auto",
        },
        Object {},
      ]
    }
  />
</View>
`;
